executor/src/witgen/generator.rs

```diff
@@ -23,7 +23,7 @@ pub struct Generator<'a, T: FieldElement, QueryCallback: Send + Sync> {
     witnesses: BTreeSet<&'a PolynomialReference>,
     machines: Vec<Box<dyn Machine<T>>>,
     query_callback: Option<QueryCallback>,
-    global_range_constraints: BTreeMap<&'a PolynomialReference, RangeConstraint<T>>,
+    global_range_constraints: BTreeMap<u64, RangeConstraint<T>>,
```

Comment:
Maybe `BTreeMap<u64, RangeConstraint<T>>` could be extracted as a type so it doesn't need to be repeated everywhere? Not too important, just a readability suggestion.

Reply:
Hmm, I actually find it more readable this way, because I don't have to remember a new type and that it's just a map. That might be a matter of taste though; I've realized that Rust folks like to introduce new types a lot 😄

Reply:
With that I actually agree. There are several places in witgen where `usize` is the key, and it's not clear whether it's supposed to be an ID or some sort of index.

After a few iterations, I figured it's best to use the existing `PolyID`. It also contains the type (committed or fixed), which is actually what we want, because the `u64` ID numbers will collide between the types!

I renamed `PolynomialReference::poly_id()` -> `PolynomialReference::poly_id_u64()` and added `PolynomialReference::poly_id()` for convenience. I'll remove `poly_id_u64()` in #412, to keep the changes of this PR small.